MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication+Internal.swift

```diff
@@ -26,17 +26,20 @@ import Foundation
 
 extension MSALNativeAuthPublicClientApplication {
 
-    func signUpUsingPasswordInternal(
+    func signUpInternal(
         username: String,
-        password: String,
+        password: String?,
         attributes: [String: Any]?,
         correlationId: UUID?
-    ) async -> MSALNativeAuthSignUpControlling.SignUpStartPasswordControllerResponse {
+    ) async -> MSALNativeAuthSignUpControlling.SignUpStartControllerResponse {
         guard inputValidator.isInputValid(username) else {
-            return .init(.error(SignUpPasswordStartError(type: .invalidUsername)))
+            return .init(.error(SignUpStartError(type: .invalidUsername)))
         }
-        guard inputValidator.isInputValid(password) else {
-            return .init(.error(SignUpPasswordStartError(type: .invalidPassword)))
+
+        if let password = password {
```

Comment:
in terms of stile here, I'd put the if let and password check in the same if block, something like:
```suggestion
        if let password = password, !inputValidator.isInputValid(password) {
            return .init(.error(SignUpStartError(type: .invalidPassword)))
        }
```

Reply:
Updated